### PR TITLE
Fix warnings/deprecations

### PR DESCRIPTION
**Pull request by MortyMerr**
As a potential contributor, I would like to see only my warnings in CI logs. However, right now there are few deprecation messages, swiftlint warnings and type-check warnings. This PR fix them all.

### CODE REVIEW

File: Tests/RxSwiftTests/Observable+MergeTests.swift [hunk 1]
```diff
@@ -1222,7 +1222,7 @@ extension ObservableMergeTest {
     func testFlatMapFirst_Complete() {
         let scheduler = TestScheduler(initialClock: 0)
 
-        let xs = scheduler.createHotObservable([
+        let xs: TestableObservable<TestableObservable<Int>> = scheduler.createHotObservable([
```
Comment (freak4pc) [inline Tests/RxSwiftTests/Observable+MergeTests.swift]: Why is this needed? Seems to work without.

File: RxCocoa/macOS/NSSlider+Rx.swift [hunk 2]
```diff
@@ -16,7 +16,7 @@ extension Reactive where Base: NSSlider {
     /// Reactive wrapper for `value` property.
     public var value: ControlProperty<Double> {
         return self.base.rx.controlProperty(
-            getter: { control in
+            getter: { control -> Double in
```
Comment (freak4pc) [inline RxCocoa/macOS/NSSlider+Rx.swift]: Why is this needed?

File: Tests/RxSwiftTests/Observable+MergeTests.swift (hunk 1 above)
Comment (MortyMerr) [reply Tests/RxSwiftTests/Observable+MergeTests.swift]: This line cause type-check time limit warning:
```
/Users/travis/build/ReactiveX/RxSwift/Tests/RxSwiftTests/Observable+MergeTests.swift:1225:28: expression took 155ms to type-check (limit: 100ms)
```
This fix prevents it. If u think this warning is ok, i am ready to discard changes and leave only deprecation fix + swiftlint disabling

File: RxCocoa/macOS/NSSlider+Rx.swift (hunk 2 above)
Comment (MortyMerr) [reply RxCocoa/macOS/NSSlider+Rx.swift]: Same as above
